examples/cdk/lib/example-function.MyFunctionWithMiddy.ts

```diff
@@ -1,43 +1,73 @@
+import { Context } from 'aws-lambda';
+import { Events } from '@aws-lambda-powertools/commons';
+import { Metrics, MetricUnits, logMetrics } from '@aws-lambda-powertools/metrics';
+import { Tracer, captureLambdaHandler } from '@aws-lambda-powertools/tracer';
+import { Logger, injectLambdaContext } from '@aws-lambda-powertools/logger';
 
-const metrics = new Metrics({ namespace: 'CDKExample', service: 'withMiddy' }); // Sets metric namespace, and service as a metric dimension
+const namespace = 'CDKExample';
+const serviceName = 'MyFunctionWithMiddyMiddleware';
 
-type CustomEvent = {
-  throw: boolean
-};
+const metrics = new Metrics({ namespace: namespace, serviceName: serviceName });
+const logger = new Logger({ logLevel: 'INFO', serviceName: serviceName });
+const tracer = new Tracer({ serviceName: serviceName });
 
-class MyFunctionWithDecorator {
+const lambdaHandler = async (event: typeof Events.Custom.CustomEvent, context: Context) => {
+  // ### Experiment with Logger
+  // AWS Lambda context is automatically injected by the middleware
 
-  @metrics.logMetrics({ captureColdStartMetric: true })
-  public handler(_event: CustomEvent, _context: Context, _callback: Callback<unknown>): void | Promise<unknown> {
-    metrics.addMetric('test-metric', MetricUnits.Count, 10);
-    if (_event.throw) {
-      throw new Error('Test error');
-    }
-  }
-}
+  logger.addPersistentLogAttributes({
+    testKey: 'testValue',
+  });
+  logger.debug('This is an DEBUG log'); // Won't show by default
+  logger.info('This is an INFO log');
+  logger.warn('This is an WARN log');
+  logger.error('This is an ERROR log');
 
-const handler = middy(async (_event, _context) => {
+  // ### Experiment with Metrics
+  // Default metrics, cold start, and throwOnEmptyMetrics are enabled by the middleware
 
-  const handlerClass = new MyFunctionWithDecorator();
+  metrics.addMetric('test-metric', MetricUnits.Count, 10);
 
-  return handlerClass.handler(_event, _context, () => console.log('Lambda invoked!'));
-});
+  const metricWithItsOwnDimensions = metrics.singleMetric();
+  metricWithItsOwnDimensions.addDimension('InnerDimension', 'true');
+  metricWithItsOwnDimensions.addMetric('single-metric', MetricUnits.Percent, 50);
+  
+  // ### Experiment with Tracer
 
-handler.before(async (_request) => {
-  metrics.addMetric('beforeHandlerCalled', MetricUnits.Count, 1);
-});
+  // Service & Cold Start annotations will be added for you by the decorator/middleware
 
-handler.after(async (_request) => {
-  // Won't be flushed since happens after 
-  metrics.addMetric('afterHandlerCalled', MetricUnits.Count, 1);
+  // These traces will be added to the main segment (## index.handler)
+  tracer.putAnnotation('awsRequestId', context.awsRequestId);
+  tracer.putMetadata('eventPayload', event);
 
-});
+  // Create another subsegment & set it as active
+  const handlerSegment = tracer.getSegment(); // This is the custom segment created by Tracer for you (## index.handler)
```

Comment:
(nit) In `MyFunction.ts`, the comment is :"// This is the facade segment (the one that is created by AWS Lambda)"

Reply:
Yep, that's right. `MyFunction.ts` is an example of manual instrumentation and as such customers need to retrieve the `facade` segment (aka the one created by AWS Lambda) before they can create the one called `## index.handler`, this is needed because customers can't annotate the main Lambda segment.

![image](https://user-images.githubusercontent.com/7353869/148516176-f44b34f5-c804-42df-bcbc-f4ac4311413f.png)
(screenshot above comes [from here](https://docs.aws.amazon.com/lambda/latest/dg/services-xray.html))

In this file instead, since we are using a Middy middleware the retrieval of the `facade` segment + the creation of the custom (`## index.hanlder`) one is done for you, so the first time you run `tracer.getSegment` you're actually getting that one, and not the `facade` one; hence the var name & comment difference.